src/processors/ipfsFile/syncExistingPinsProcessor.ts

```diff
@@ -11,29 +12,80 @@ const ipfsFilesOutOfSyncWithPins: (field: TrackMediaField) => Trigger<undefined>
     select distinct "${field}IPFSHash", "${field}URL" from "${Table.processedTracks}" as t
     left outer join "${Table.ipfsFiles}" as file
     on t."${field}IPFSHash" = file.cid
-    where
+    where (
       t."${field}IPFSHash" is not null and
       file.cid is null and
       file.error is null and
       t."${field}URL" ILIKE ('%' || t."${field}IPFSHash" || '%')
+    ) or (
+      t."${field}IPFSHash" is not null and
+      t."${field}URL" is not null and
+      file.cid is null and
+      file.error is null
+    )
   `)
 
    return tracksWithCIDsMatchingURL.rows;
  }
 
+const ipfsFilesPinnedWithoutUrl: (field: TrackMediaField) => Trigger<undefined> =
+ (field) => async (clients) => {
+   const tracksWithCIDWithoutURL = await clients.db.rawSQL(`
+    select "id", "${field}IPFSHash", "${field}URL" from "${Table.processedTracks}" as t
+    left outer join "${Table.ipfsFilesUrls}" as url
+    on t."${field}URL" = url.url
+    where
+      t."${field}IPFSHash" is not null and
+      t."${field}URL" is null and
+      url.cid is null and
+      url.error is null
+  `)
+
+   return tracksWithCIDWithoutURL.rows;
+ }
+
 export const ipfsFileSyncExistingPinsProcessor: (field: TrackMediaField) => Processor =
```

Comment:
change to ipfsFileSyncExistingUploadsProcessor rather than pins (and same for the trigger name/console.log to avoid confusion)